pg_shard--1.1.sql

```diff
@@ -140,3 +140,51 @@ $sync_table_metadata_to_citus$ LANGUAGE 'plpgsql';
 
 COMMENT ON FUNCTION sync_table_metadata_to_citus(text)
 		IS 'synchronize a distributed table''s pg_shard metadata to CitusDB';
+
+-- Prepares a specified distributed table to work with COPY operations by
+-- creating a temporary table and trigger to handle rows coming from a COPY.
+-- After each INSERT, a provided sequence is incremented to track the total -- number of copied rows.
+CREATE FUNCTION prepare_distributed_table_for_copy(relation regclass, sequence regclass) RETURNS void
+AS $pdtfc$
+	DECLARE
+		table_name text;
+		temp_table_name text;
+		attr_names text[];
+		attr_list text;
+		param_list text;
+		using_list text;
+		insert_command text;
+		func_tmpl CONSTANT text  := $$  CREATE FUNCTION pg_temp.copy_to_insert() RETURNS trigger
+										AS $cti$
+											BEGIN
+												EXECUTE %L USING %s;
+												PERFORM nextval(%L);
+												RETURN NULL;
+											END;
+										$cti$ LANGUAGE plpgsql VOLATILE;
+									$$;
+		table_tmpl CONSTANT text := 'CREATE TEMPORARY TABLE %I (LIKE %s)';
```

Comment:
I guess I had a style that I used normal quotes if it fit on one line and $$ for multi-line things. Kind of like how you wouldn’t use a here doc for every string declaration in a shell script…

—Jason

> On Mar 10, 2015, at 3:29 PM, sumedhpathak <notifications@github.com> wrote:
> 
> In pg_shard--1.1.sql <https://github.com/citusdata/pg_shard/pull/82#discussion_r26168240>:
> 
> > +		temp_table_name text;
> > +		attr_names text[];
> > +		attr_list text;
> > +		param_list text;
> > +		using_list text;
> > +		insert_command text;
> > +		func_tmpl CONSTANT text  := $$  CREATE FUNCTION pg_temp.copy_to_insert() RETURNS trigger
> > +										AS $cti$
> > +											BEGIN
> > +												EXECUTE %L USING %s;
> > +												PERFORM nextval(%L);
> > +												RETURN NULL;
> > +											END;
> > +										$cti$ LANGUAGE plpgsql VOLATILE;
> > +									$$;
> > +		table_tmpl CONSTANT text := 'CREATE TEMPORARY TABLE %I (LIKE %s)';
> Why does this statement use quotes instead of $$? Could we be consistent?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/citusdata/pg_shard/pull/82/files#r26168240>.
>